Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -2113,7 +2113,13 @@ extension BrowserViewController: ContextMenuHelperDelegate {
             let copyAction = UIAlertAction(title: copyImageTitle, style: UIAlertActionStyle.Default) { (action: UIAlertAction) -> Void in
                 let pasteBoard = UIPasteboard.generalPasteboard()
                 pasteBoard.string = url.absoluteString
-                // TODO: put the actual image on the clipboard
+                // put the actual image on the clipboard
+                // do this be asyncronously just in case we're in a low bandwidth situation
+                // Is this neccessary, or is the image already sat in cache?
+                NSURLSession.sharedSession().dataTaskWithURL(url, completionHandler: { (data, _, error) -> Void in
```

Comment:
OK. I'll keep what we have here. Currently under good network conditions the copy feels instantaneous as by the time you've navigated to another app and pressed paste then the image has downloaded.

Under bad network conditions with large files, you'll end up pasting a link. This is better than nothing and what we have currently. We can revisit the JS option if we feel that we need an instantaneous solution.